### PR TITLE
Add retag and push script

### PR DESCRIPTION
**Pull request by erkannt**
Adds  `push-image.sh` from [libero/reviewer-submission](https://github.com/libero/reviewer-submission) as `retag-and-push.sh`.

- `push-image.sh` has been sucessfully used in reviewer-submission to push both master-sha and semver images
- add testing via a bats script

### CODE REVIEW

**Comment by diversemix** [comment]
Locally I get "bash: shellcheck: command not found"

**Comment by erkannt** [comment]
> Locally I get "bash: shellcheck: command not found"

You can install it with `apt install shellcheck`. Should I add this to the readme?

**Review by diversemix — APPROVED** [review]

**Comment by diversemix** [comment]
> > Locally I get "bash: shellcheck: command not found"
> 
> You can install it with `apt install shellcheck`. Should I add this to the readme?

Yeah - you could also check in the Makefile?

File: tests/github-retag-and-push.bats [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    [ "$status" -eq 0 ]
+    run docker pull ${DOCKER_REGISTRY}liberoadmin/my-dummy-project:1
+    [ "$status" -eq 0 ]
+    run docker pull ${DOCKER_REGISTRY}liberoadmin/my-dummy-project:1.2
+    [ "$status" -eq 0 ]
+    run docker pull ${DOCKER_REGISTRY}liberoadmin/my-dummy-project:1.2.43
+    [ "$status" -eq 0 ]
+}
+
+@test "missing v in tag" {
```
Comment (giorgiosironi) [inline tests/github-retag-and-push.bats:64]: did we specify the `v` requirement somewhere? No preference on the choice personally, but [the only existing tags](https://github.com/libero/event-bus) I can find don't use it.

Comment (thewilkybarkid) [reply tests/github-retag-and-push.bats:64]: I normally use it (https://github.com/libero/php-coding-standard/releases).

**Comment by thewilkybarkid** [comment]
Can we switch to using the [Docker action](https://github.com/docker/build-push-action)? Doesn't seem that this does anything not possible there?

**Comment by giorgiosironi** [comment]
I remember the Docker action was evaluated but discarded, I'll leave to @erkannt the longer explanation.

File: tests/github-retag-and-push.bats (hunk 1 above)
Comment (erkannt) [reply tests/github-retag-and-push.bats:64]: The ADR uses `v` in the example, but does not specify its use.  In my opinion this is an oversight.  As the script relies on `GITHUB_REF` we need to agree on a convention regarding this.

**Comment by erkannt** [comment]
> Can we switch to using the [Docker action](https://github.com/docker/build-push-action)? Doesn't seem that this does anything not possible there?

Indeed we could. In my research I found a plethora of actions including the official docker one. Sadly none of the ones I looked at conformed to our ADR. In the long run we could bundle the script into a github action of our own. But for now I feel that this is overkill due to us having to load the submodule in most pipelines anyway.

**Comment by thewilkybarkid** [comment]
> Sadly none of the ones I looked at conformed to our ADR.

Is this written up somewhere?

(This shouldn't hold up this PR, just wondering what needs to change to move to the official one in the future.)

File: tests/github-retag-and-push.bats (hunk 1 above)
Comment (giorgiosironi) [reply tests/github-retag-and-push.bats:64]: Proposing to edit the ADR (but only because it's in a `Proposed` state still) to mandate the `v`.

**Comment by erkannt** [comment]
> > Sadly none of the ones I looked at conformed to our ADR.
> 
> Is this written up somewhere?
> 
> (This shouldn't hold up this PR, just wondering what needs to change to move to the official one in the future.)

 I haven't written this up. What would be a good place to do this? Or would this maybe be a good topic for a infrastructure office hour on github actions?

File: tests/github-retag-and-push.bats (hunk 1 above)
Comment (erkannt) [reply tests/github-retag-and-push.bats:64]: Opened a WIP PR for this https://github.com/libero/community/pull/38/files

File: Makefile [hunk 2]
```diff
@@ -5,6 +5,8 @@ setup: install-deps launch-registry
 install-deps:
 	npm install bats@1.x
 	docker pull busybox
+	command -v shellcheck || echo "  please run :	'apt install shellcheck'"
```
Comment (thewilkybarkid) [inline Makefile]: Since it'd be `brew install shellcheck` on a Mac, should use https://hub.docker.com/r/koalaman/shellcheck?

Comment (erkannt) [reply Makefile]: The github runner comes with shellcheck, so an image isn't needed there. To keep things consistent I have instead adjusted the readme and error message wording to be more os agnostic.

File: Makefile [hunk 3]
```diff
@@ -5,6 +5,8 @@ setup: install-deps launch-registry
 install-deps:
 	npm install bats@1.x
 	docker pull busybox
+	command -v shellcheck || echo "  please apt/brew install shellcheck"
```
Comment (giorgiosironi) [inline Makefile:8]: not super reproducible, but so far have been relying on the OS-installed one

Comment (erkannt) [reply Makefile:8]: Given the scope of this repo (currently two scripts) I would propose that we merge this nonetheless. We also are not installing npm or docker with the makefile.

**Review by giorgiosironi — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (giorgiosironi) [reply Makefile:8]: Practically the rate of change on shellcheck is also quite low, [one or two versions per year](https://github.com/koalaman/shellcheck/tags). Approved.

**Pull request merged**